src/Types/Node.php

```diff
@@ -62,4 +65,34 @@ public function id(): int
     {
         return $this->id;
     }
+
+    public function property(string $key)
+    {
+        if ($this->properties->hasKey($key)) {
+            return $this->properties->get($key);
+        }
+    }
+
+    private function mapProperties(Map $properties): Map
+    {
+        return $properties->map(function($key, $value) {
+            if (is_object($value)) {
+                switch(get_class($value)) {
+                    case BoltDate::class:
+                        return Date::makeFromBoltDate($value);
+                    case BoltDuration::class:
+                        return Duration::makeFromBoltDuration($value);
+                    case BoltDateTime::class:
+                        return DateTime::makeFromBoltDateTime($value);
+                }
+            }
```

Comment:
Repetition detected here, will probably export into an external "factory" type of mapper.

Reply:
My only concern with a dictionary approach is the difference between Bolt and HTTP responses. i.e. in Bolt the value is a `Node` instance from Bolt driver, and the type from HTTP is `'node'` string. It's illustrated in the implementations of

```php
Laudis\Neo4j\Formatter\OGMFormatter::buildResult
// and
Laudis\Neo4j\Formatter\OGMFormatter::mapValue
```